### DISCUSSION TITLE
Create new workspace when switching to next one

### DISCUSSION THREAD

**Discussion by Drexxl-the-Walrus** [i3/i3#5853] [Q&A]
Hello all!

I have been using i3 for close to a year and I am loving it, now, on to my question.

I use a command to switch between next and previous workspace. Is there a command with this function that, if I am at my "rightmost" workspace, creates a new one after it instead of switching back to the first?
Example: I have workspaces 1,2,3. If i am at workspace 2, the command will move me to workspace 3. If I am at workspace 3, it will create workspace 4.
I have not been able to find anything in the i3 guide, but I am quite bad at using it.

Thanks in advance!

**Comment by mikeredev** [comment]
I dont think it will create any workspaces that aren't already defined in your config (i.e., you need to set $ws4 before it will open). you can use `i3-msg 'workspace x'` directly to open workspace x but it won't have any rules applied to it

**Comment by orestisfl** [comment]
I use this script to open the next workspace number after the currently focused one. E.g. if you have workspace number 5 focused and numbers 1,2,6,7,9 exist it will create and focus workspace number 8. You could adjust it to your needs.

```python
#!/usr/bin/python
import json
from subprocess import check_output

workspaces = json.loads(check_output(["i3-msg", "-t", "get_workspaces"]))
workspace_nums = [ws["num"] for ws in workspaces]
f = next(ws["num"] for ws in workspaces if ws["focused"])

for next_num in range(f + 1, 50):
    if next_num not in workspace_nums:
        print(next_num)
        break
```

config:
```
bindsym $mod+o exec --no-startup-id i3-msg workspace number $($HOME/bin/i3-next-available-workspace.py)
```
